ocaml/xapi/xapi_vm.ml

```diff
@@ -221,6 +221,16 @@ let unpause ~__context ~vm =
 let set_xenstore_data ~__context ~self ~value =
   Xapi_xenops.set_xenstore_data ~__context ~self value
 
+(* CP-18860: check memory limits if using nested_virt on start *)
+let assert_memory_constraints ~__context ~vm platformdata =
```

Comment:
I think it would make sense to move this into xapi_vm_helpers.ml and call it from `assert_can_boot_here`... but perhaps make it more general, basing it on `Xapi_vm_memory_constraints.Vm_memory_constraints.assert_valid_for_current_context` without checking for nested-virt here.